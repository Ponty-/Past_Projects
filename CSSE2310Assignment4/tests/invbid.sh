#!/bin/bash

if [ $# != 4 ]
then
    echo "Usage: $0 binarydir port resultdir correct_dir" 
    exit 1
fi



bindir=$1
port=$2
res=$3
right=$4
name=$0

expected=0

if [ ! -f $bindir/client499 ]
then
    echo "$bindir/client499 is missing"
    exit 1
fi

if [ ! -f $bindir/serv499 ]
then
    echo "$bindir/serv499 is missing"
    exit 1
fi



$bindir/serv499 $port .. 1.deck &
spid=$!
$bindir/client499 A game $port > $res/$name.a.out 2>$res/$name.a.err < iamoves &
pid1=$!
$bindir/client499 B game $port > $res/$name.b.out 2>$res/$name.b.err < ibmoves &
pid2=$!
$bindir/client499 C game $port > $res/$name.c.out 2>$res/$name.c.err < icmoves &
pid3=$!
$bindir/client499 D game $port > $res/$name.d.out 2>$res/$name.d.err < idmoves &
pid4=$!

wait $pid1 $pid2 $pid3 $pid4 2> /dev/null
kill $spid 2> /dev/null
wait $spid 2> /dev/null

#if [ $result != $expected ]
#then
#    echo "Expected $expected got $result"
#    exit 1
#fi

for v in a b c d
do

diff -q $res/$name.$v.out $right/$name.$v.out > /dev/null
if [ $? != 0 ]
then
   echo "Stdout mismatch for client $v"
   echo "diff $res/$name.$v.out $right/$name.$v.out"
   exit 2
fi

diff -q $res/$name.$v.err $right/$name.$v.err > /dev/null
if [ $? != 0 ]
then
   echo "Stderr mismatch for client $v"
   echo "diff $res/$name.$v.err $right/$name.$v.err"
   exit 3
fi

done

echo OK
exit 0

