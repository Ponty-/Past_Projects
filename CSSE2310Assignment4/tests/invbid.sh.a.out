Info: ..
Info: Team1: A, C
Info: Team2: B, D
S: Q 7
C: Q 7 3
D: K T 7 5
H: A T 6 2
Bid> Bid> Bid> Info: B passes
Info: C bids 9H
Info: C plays AD
Info: D plays AS
S: Q 7
C: Q 7 3
D: K T 7 5
H: A T 6 2
[D] play> Info: B plays 3D
Info: C won
Info: C plays QD
Info: D plays KS
S: Q 7
C: Q 7 3
D: T 7 5
H: A T 6 2
[D] play> Info: B plays 6D
Info: C won
Info: C plays 8D
Info: D plays JS
S: Q 7
C: Q 7 3
D: 7 5
H: A T 6 2
[D] play> Info: B plays 9D
Info: B won
Info: B plays 6S
Info: C plays 9S
Info: D plays 8S
S: Q 7
C: Q 7 3
D: 7
H: A T 6 2
[S] play> Info: C won
Info: C plays KC
Info: D plays JC
S: Q
C: Q 7 3
D: 7
H: A T 6 2
[C] play> Info: B plays 2C
Info: C won
Info: C plays JH
Info: D plays 3H
S: Q
C: Q 7
D: 7
H: A T 6 2
[H] play> Info: B plays 4H
Info: C won
Info: C plays 4S
Info: D plays 3S
S: Q
C: Q 7
D: 7
H: A T 6
[S] play> Info: B plays TS
Info: A won
S:
C: Q 7
D: 7
H: A T 6
Lead> Info: B plays JD
Info: C plays 4D
Info: D plays 5S
Info: B won
Info: B plays AC
Info: C plays 9C
Info: D plays 8C
S:
C: Q 7
D:
H: A T 6
[C] play> Info: B won
Info: B plays TC
Info: C plays 4C
Info: D plays 5C
S:
C: Q
D:
H: A T 6
[C] play> Info: A won
S:
C:
D:
H: A T 6
Lead> Info: B plays 9H
Info: C plays 8H
Info: D plays 7H
Info: B won
Info: B plays 6C
Info: C plays 2D
Info: D plays 2S
S:
C:
D:
H: A T
[C] play> Info: A won
S:
C:
D:
H: T
Lead> Info: B plays KH
Info: C plays 5H
Info: D plays QH
Info: B won
Info: Team 1=-300, Team 2=0
S: Q 7
C: Q 7 3
D: K T 7 5
H: A T 6 2
Bid> S: Q 7
C: Q 7 3
D: K T 7 5
H: A T 6 2
Lead> Info: B plays 3D
Info: C plays AD
Info: D plays 2S
Info: C won
Info: C plays JH
Info: D plays QH
S: Q 7
C: Q 7 3
D: T 7 5
H: A T 6 2
[H] play> Info: B plays 4H
Info: D won
Info: D plays AS
S: Q 7
C: Q 7 3
D: T 7 5
H: A T 6
[S] play> Info: B plays TS
Info: C plays 9S
Info: D won
Info: D plays KS
S: 7
C: Q 7 3
D: T 7 5
H: A T 6
[S] play> Info: B plays 6S
Info: C plays 4S
Info: D won
Info: D plays JS
S:
C: Q 7 3
D: T 7 5
H: A T 6
[S] play> Info: B plays AC
Info: C plays KC
Info: D won
Info: D plays 8S
S:
C: 7 3
D: T 7 5
H: A T 6
[S] play> Info: B plays TC
Info: C plays 9C
Info: D won
Info: D plays 5S
S:
C: 7
D: T 7 5
H: A T 6
[S] play> Info: B plays 6C
Info: C plays 4C
Info: D won
Info: D plays 3S
S:
C:
D: T 7 5
H: A T 6
[S] play> Info: B plays 2C
Info: C plays QD
Info: D won
Info: D plays JC
S:
C:
D: T 7
H: A T 6
[C] play> Info: B plays JD
Info: C plays 8D
Info: D won
Info: D plays 8C
S:
C:
D: 7
H: A T 6
[C] play> Info: B plays 6D
Info: C plays 4D
Info: D won
Info: D plays 5C
S:
C:
D:
H: A T 6
[C] play> Info: B plays 9D
Info: C plays 2D
Info: A won
S:
C:
D:
H: T 6
Lead> Info: B plays KH
Info: C plays 8H
Info: D plays 7H
Info: B won
Info: B plays 9H
Info: C plays 5H
Info: D plays 3H
S:
C:
D:
H: T
[H] play> Info: A won
Info: Team 1=-600, Team 2=0
Info: Winner is Team 2
